app/src/main/java/co/omise/android/example/PaymentSetting.kt

```diff
@@ -100,6 +102,17 @@ object PaymentSetting {
                 }
                 .filterNotNull()
 
+        val tokenizationMethods = paymentMethodPreferences
+                .filter { it.value }
+                .toMap()
+                .map {
+                    when (it.key) {
+                        context.getString(R.string.payment_preference_googlepay_key) -> TokenizationMethod.GooglePay
+                        else -> null
+                    }
+                }
+                .filterNotNull()
```

Comment:
Not sure if it works but maybe `mapNotNull` can be used for combining `.map` and `.filterNotNull`.